byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
> This benchmark looks suspect. I'd never loop over floating point numbers that way. If, for example, you replaced double with float, the loop won't terminate anymore due to cancellation for i >= 2.

Well, give me another benchmark that measures the same thing. I already observed this fact several times in different contexts. And, btw, I get similar results with my other linux machine (Core i5-4300 @ 1.9 GHz, glibc 2.24).

The reason why it is slower is **not** the difference of performance of FP instructions. The reason is that the approximation algorithm used by libc needs to be more involved to get the extra precision.

> Just do some basic arithmetic operations in a loop. There is no difference on my machine between float and double performance in that case.

This is unrelated to this PR, but in practice, if I just do some basic arithmetic operations in a loop, there are some chances that it will be compiled to SIMD instructions, leading to a larger throughput.